src/app/util/ember-compatibility-functions.cpp

```diff
@@ -1039,3 +1039,22 @@ void MatterReportingAttributeChangeCallback(const ConcreteAttributePath & aPath)
 {
     return MatterReportingAttributeChangeCallback(aPath.mEndpointId, aPath.mClusterId, aPath.mAttributeId);
 }
+
+void MatterReportingAttributeChangeCallback(EndpointId endpoint)
+{
+    // Attribute writes have asserted this already, but this assert should catch
+    // applications notifying about changes from their end.
+    assertChipStackLockedByCurrentThread();
+
+    ClusterInfo info;
+    info.mEndpointId = endpoint;
+
+    // We are adding or enabling a whole endpoint, in this case, we do not touch the cluster data version.
+
+    InteractionModelEngine::GetInstance()->GetReportingEngine().SetDirty(info);
+
+    // Schedule work to run asynchronously on the CHIP thread. The scheduled work won't execute until the current execution context
```

Comment:
Please update to tip.  SetDirty() handles all that itself on tip.

Reply:
Thanks for notice.